### PR TITLE
[ONNX] Fix for batchnorm training op mode (#52758)

### PR DESCRIPTION
**Pull request by BowenBao**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #63249 Enhance shape (#60335)
* #62765 [ONNX] Suppport torch.dot and torch.nn.utils.spectral_norm (#62596)
* #62764 [ONNX] Update repeat_interleave for dynamic repeats (#59979)
* #62763 [ONNX] Fix an issue that optimizations might adjust graph inputs unexpectedly. (#61280)
* #62762 [ONNX] Fix controlflow shape inference with contrib op (#60707)
* **#62760 [ONNX] Fix for batchnorm training op mode (#52758)**
* #62759 [ONNX] Remove aten parameter (#61652)
* #62758 [ONNX] Add support for opset14 in PT-ONNX exporter (#59486)
* #62757 [ONNX] Support lstm_cell symbolic (#61476)

* Rebase

# Conflicts:
#	torch/csrc/jit/passes/onnx/eval_peephole.cpp

# Conflicts:
#	test/onnx/test_utility_funs.py
#	torch/onnx/symbolic_opset9.py

* Update symbolic_opset12.py

* Update test.sh
# Conflicts:
#	.jenkins/caffe2/test.sh

* Merge

* Fix utility tests

# Conflicts:
#	test/onnx/test_pytorch_onnx_onnxruntime.py
#	test/onnx/test_utility_funs.py

* Fix for comment

* Enable BN tests

* Fix for test

* Update test_pytorch_onnx_onnxruntime.py

* Update test_pytorch_onnx_onnxruntime.py

* Update test_utility_funs.py

* Update test_pytorch_onnx_onnxruntime.py

Differential Revision: [D30349060](https://our.internmc.facebook.com/intern/diff/D30349060)

### CODE REVIEW

**Comment by msaroufim** [comment]
@msaroufim has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30349060).

File: .jenkins/caffe2/test.sh [hunk 1]
```diff
@@ -170,7 +170,9 @@ if [[ "$BUILD_ENVIRONMENT" == *onnx* ]]; then
   # JIT C++ extensions require ninja, so put it into PATH.
   export PATH="/var/lib/jenkins/.local/bin:$PATH"
   if [[ "$BUILD_ENVIRONMENT" == *py3* ]]; then
-    pip install -q --user onnxruntime==1.8.0
+    pip install -q --user flatbuffers==2.0
+    wget https://ortpypackage.blob.core.windows.net/ort-nightly/ort_nightly-1.8.0.dev202107131-cp36-cp36m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
+    pip install -q --user ort_nightly-1.8.0.dev202107131-cp36-cp36m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
```
Comment (malfet) [inline .jenkins/caffe2/test.sh:175]: Why not combine those two lines? Also, please add comment to PR description that version of onnxruntime is being updated to nightly and why it is necessary to do so.
```suggestion
    pip install -q --user https://ortpypackage.blob.core.windows.net/ort-nightly/ort_nightly-1.8.0.dev202107131-cp36-cp36m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl```

**Review by malfet — CHANGES_REQUESTED** [review]
Please remove uninformative "Conflict resolved" lines from PR description and mention that it also updates onnxruntime dependency. 
Also, it would be good to have a bit more insight into what "Fix for batchnorm training op mode" means
